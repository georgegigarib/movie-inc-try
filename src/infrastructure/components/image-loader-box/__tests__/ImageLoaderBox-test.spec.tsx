import React from 'react';
import { render, fireEvent, waitFor, act } from '@testing-library/react-native';
import ImageWithLoader from '@/src/infrastructure/components/image-loader-box/ImageLoaderBox';

jest.useFakeTimers();

describe('ImageWithLoader', () => {
  it('shows loading indicator initially', () => {
    const { getByTestId, toJSON } = render(
      <ImageWithLoader source={{ uri: 'https://example.com/image.jpg' }} testID="image-loader" />
    );
    expect(getByTestId('image-loader')).toBeTruthy();
    expect(getByTestId('loading-container')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('shows error message if loading times out', async () => {
    const { getByTestId, toJSON } = render(
      <ImageWithLoader source={{ uri: 'https://example.com/image.jpg' }} testID="image-loader" />
    );

    act(() => {
      jest.advanceTimersByTime(15000);
    });

    await waitFor(() => {
      expect(getByTestId('image-loader')).toBeTruthy();
      expect(getByTestId('error-container')).toBeTruthy();
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('shows image and hides loading indicator when image loads', async () => {
    const { getByTestId, queryByTestId, toJSON } = render(
      <ImageWithLoader source={{ uri: 'https://example.com/image.jpg' }} testID="image-loader" />
    );

    act(() => {
      const image = getByTestId('image-loader-image');
      fireEvent(image, 'load');
      jest.advanceTimersByTime(700);
    });

    await waitFor(() => {
      expect(queryByTestId('loading-container')).toBeNull();
      expect(getByTestId('image-loader-image')).toBeTruthy();
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('shows error message if image fails to load', async () => {
    const { getByTestId, queryByTestId, toJSON } = render(
      <ImageWithLoader source={{ uri: 'https://example.com/image.jpg' }} testID="image-loader" />
    );

    act(() => {
      const image = getByTestId('image-loader-image');
      fireEvent(image, 'error');
    });

    await waitFor(() => {
      expect(queryByTestId('loading-container')).toBeNull();
      expect(getByTestId('error-container')).toBeTruthy();
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
