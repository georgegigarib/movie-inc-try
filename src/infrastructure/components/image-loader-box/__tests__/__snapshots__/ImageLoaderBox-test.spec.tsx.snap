// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageWithLoader shows error message if image fails to load 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
  testID="image-loader"
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.8)",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
    testID="error-container"
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 12,
            "fontWeight": "300",
          },
          undefined,
        ]
      }
    >
      Picture not available
    </Text>
  </View>
</View>
`;

exports[`ImageWithLoader shows error message if loading times out 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
  testID="image-loader"
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.8)",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
    testID="error-container"
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 12,
            "fontWeight": "300",
          },
          undefined,
        ]
      }
    >
      Picture not available
    </Text>
  </View>
</View>
`;

exports[`ImageWithLoader shows image and hides loading indicator when image loads 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
  testID="image-loader"
>
  <View
    collapsable={false}
    style={
      {
        "opacity": 0,
      }
    }
  >
    <Image
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      source={
        {
          "uri": "https://example.com/image.jpg",
        }
      }
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
      testID="image-loader-image"
    />
  </View>
</View>
`;

exports[`ImageWithLoader shows loading indicator initially 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
  testID="image-loader"
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.8)",
          "bottom": 0,
          "justifyContent": "center",
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
      ]
    }
    testID="loading-container"
  >
    <ActivityIndicator
      color="#b6b6b6"
      size="large"
    />
  </View>
  <View
    collapsable={false}
    style={
      {
        "opacity": 0,
      }
    }
  >
    <Image
      onError={[Function]}
      onLoad={[Function]}
      onLoadStart={[Function]}
      source={
        {
          "uri": "https://example.com/image.jpg",
        }
      }
      style={
        {
          "height": "100%",
          "width": "100%",
        }
      }
      testID="image-loader-image"
    />
  </View>
</View>
`;
